### PR TITLE
GitHub issue#31057 add haproxy.router.openshift.io/set-forwarded-headers to annotation table

### LINKED ISSUE #31057: [enterprise-4.6] Issue in file networking/routes/route-configuration.adoc
### Which section(s) is the issue in?
Route-specific annotations

### What needs fixing?

missing annotation in table - haproxy.router.openshift.io/set-forwarded-headers

https://docs.openshift.com/container-platform/4.6/networking/ingress-operator.html#nw-using-ingress-forwarded_configuring-ingress (while this is linked to in the release notes, it is not linked anywhere else, and doesn't seem to come up in search)

### PR DESCRIPTION
**Pull request by bmcelvee**
Resolves https://github.com/openshift/openshift-docs/issues/31057.

Direct preview link: https://deploy-preview-31191--osdocs.netlify.app/openshift-enterprise/latest/networking/routes/route-configuration.html#nw-route-specific-annotations_route-configuration

### CODE REVIEW

**Comment by bmcelvee** [comment]
@quarterpin would you mind reviewing this PR, please? Thanks!

**Comment by quarterpin** [comment]
Imho, the statement can be re-written to explain the available options more clearly:

`haproxy.router.openshift.io/set-forwarded-headers`: sets the policy for handling the `Forwarded` and `X-Forwarded-For` HTTP headers per route. It takes any one of the below values :
`append`: appends the header, preserving any existing header.
`replace`: sets the header, removing any existing header.
`if-none`: never sets the header, but preserves any existing header.
`replace`: sets the header if it is not already set. 

This is set to `append` policy by default.

**Comment by bmcelvee** [comment]
@openshift/team-documentation PTAL, thank you!

File: modules/nw-route-specific-annotations.adoc [hunk 1]
```diff
@@ -41,6 +41,17 @@ and "-". The default is the hashed internal key name for the route. |
...
+
+|`haproxy.router.openshift.io/set-forwarded-headers` | Sets the policy for handling the `Forwarded` and `X-Forwarded-For` HTTP headers per route. The values are:
+
+`append`: appends the header, preserving any existing header. This is the default value.
+
+`replace`: sets the header, removing any existing header.
+
+`if-none`: never sets the header, but preserves any existing header.
+
+`replace`: sets the header if it is not already set.| `ROUTER_SET_FORWARDED_HEADERS`
```
Comment (jeana-redhat) [inline modules/nw-route-specific-annotations.adoc:53]: Seems a bit odd to have `replace` in here twice with different definitions. I assume it does both, depending on if there is an existing header or not, but would it work to combine them in that case?

**Review by jeana-redhat — APPROVED** [review]
Just a question about the content, but format, etc. LGTM

File: modules/nw-route-specific-annotations.adoc (hunk 1 above)
Comment (bmcelvee) [reply modules/nw-route-specific-annotations.adoc:53]: Yikes! I did something wonky. Fixing now.

Comment (bmcelvee) [reply modules/nw-route-specific-annotations.adoc]: Fixed!

**Pull request merged**